lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -167,6 +178,35 @@ WasmBinaryReader::ReadFromModule()
      return wnLIMIT;
 }
 
+SectionCode
+WasmBinaryReader::SectionHeader()
+{
+    UINT len;
+    UINT sectionSize;
+    UINT idSize;
+
+    sectionSize = LEB128(len);
+    m_moduleState.count += len;
+    idSize = LEB128(len);
+    m_moduleState.count += len;
+    const char *sectionName = (char*)(m_pc);
+    m_pc += idSize;
+    m_moduleState.count += idSize;
+
+    auto cmp = [&sectionName, &idSize](const char* n) { return !strncmp(n, sectionName, idSize); };
```

Comment:
Hmm, where are you seeing this? The section names are unrestricted and may contain null chars, as is the case with function names too, which we have previously assumed to be UTF8. This comparison should have already been rewritten to use memcmp.

From: Nagy Mostafa [mailto:notifications@github.com]
Sent: Friday, March 4, 2016 3:32 PM
To: Microsoft/ChakraCore <ChakraCore@noreply.github.com>
Cc: Kuan, George <george.kuan@intel.com>
Subject: Re: [ChakraCore] Quick implementation of section and module headers (#448)


In lib/WasmReader/WasmBinaryReader.cpp<https://github.com/Microsoft/ChakraCore/pull/448#discussion_r55104727>:

> +SectionCode

> +WasmBinaryReader::SectionHeader()

> +{

> +    UINT len;

> +    UINT sectionSize;

> +    UINT idSize;

> +

> +    sectionSize = LEB128(len);

> +    m_moduleState.count += len;

> +    idSize = LEB128(len);

> +    m_moduleState.count += len;

> +    const char *sectionName = (char*)(m_pc);

> +    m_pc += idSize;

> +    m_moduleState.count += idSize;

> +

> +    auto cmp = [&sectionName, &idSize](const char* n) { return !strncmp(n, sectionName, idSize); };

Is the encoding of Section names utf-8 ? I couldn't find this in the spec doc.

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/ChakraCore/pull/448/files#r55104727>.